### ISSUE TITLE
Allow multiple icommands versions to be installed

### ISSUE THREAD

**Issue by kript** [irods/irods#7791]
We often find ourselves in the situation where we run one version of iRODS in production, and another in test. Currently only one version of the icommands can be installed at a time, and since run-in-place is no longer supported or production, we can't build them ourselves and place in, say `/opt/renci/4.2.7` etc...

It would be very helpful if the icommands could be installed in a way that is version specific, e.g. `/usr/bin/icommands_4.1.12/ils`, `/usr/bin/icommands_4.2.7/ils`  (or even preferably `/opt/renci/4.2.7/` )etc, and have a way in one's irods_environemnt.json file to set the version required (with, say, the latest being the default).

This is probably related to irods/irods_client_icommands#66 

Thank you!

**Comment by alanking** [comment]
I believe the effort by @SwooshyCueb to provide the userspace tarball packaging option in the iCommands takes care of this? See https://github.com/irods/irods/issues/5082 and https://github.com/irods/irods_client_icommands/?tab=readme-ov-file#userspace-packaging

**Comment by SwooshyCueb** [comment]
I'm not sure I'm understanding what's being asked for here. I see mentions of headers, which are not part of the userspace tarball packages.

The tarball packager almost certainly needs some updating at this point, though.

**Comment by alanking** [comment]
Okay. I'm going to bump this out since I don't think it is critical for anybody at the moment and sat dormant for over 4 years.

**Comment by kript** [comment]
It's still very much a 'nice to have'. We've worked around it for now by using Singularity containers running the icommands but where this gets tricky is for clients built on top of the headers, e.g. [baton](https://github.com/wtsi-npg/baton).

Things have improved now the irods-devel, irods-runtime and irods-icommands are separate packages (however this still doesn't allow us to have two versions installed), but there is still the need, overall, to have at least two versions of icommands -one for production and one for testing the next version.

Hope that helps clarify,

John

**Comment by trel** [comment]
Yes, sounds like non-package install is still the easiest thing here.

`update-alternatives' is interesting, but seems ... fraught.

**Comment by mcv21** [comment]
`update-alertnatives` is quite nice really - you get to set a sensible default amongst alternatives, and then the sysadmin can override if they see fit.

**Comment by trel** [comment]
is there an rpm-world equivalent?

**Comment by mcv21** [comment]
I think there is `update-alertnatives` available for rpm-based systems (RPMFind has hits, and there's [SLES documentation](https://documentation.suse.com/sled/15-SP5/html/SLED-all/cha-update-alternative.html)), although I don't work with any rpm-based distros any more.
